src/tmpfiles/tmpfiles.c

```diff
@@ -557,6 +557,49 @@ static DIR* opendir_nomod(const char *path) {
         return xopendirat_nomod(AT_FDCWD, path);
 }
 
+static int opendir_and_stat(
+                const char *path,
+                DIR **ret,
+                struct statx *ret_sx,
+                bool *ret_mountpoint) {
+
+        _cleanup_closedir_ DIR *d = NULL;
+        STRUCT_STATX_DEFINE(sx);
+        int r;
+
+        assert(path);
+        assert(ret);
+        assert(ret_sx);
+        assert(ret_mountpoint);
+
+        d = opendir_nomod(path);
+        if (!d)
+                return log_full_errno(IN_SET(errno, ENOENT, ENOTDIR) ? LOG_DEBUG : LOG_ERR,
```

Comment:
hmpf, I am not a fan of functions that log about some errors above LOG_DEBUG but not about others...

maybe catch these errors and return success in their case, and zero out `*ret_sx` and `ret_mountpoint` in that case.

(at the very least this needs some comment that makes clear this is a bit special here.)

Reply:
also return == 0 if file is missing but things are OK, and > 0 if file was found and opened

Reply:
(also would have the benefit you wouldn't have to list the same set of error codes twice, as you do now)

Reply:
OK, I reworked it like that. It's a bit better in some areas and a worse in others. But it does avoid the repetition of the errno codes.